### PR TITLE
Test: Checks for deadlocks panics in logs per each test.

### PR DESCRIPTION
**Pull request by eloycoto**
This commit make sure that there is no panic or deadlocks in cilium logs
after executed the test.

The test uses `CurrentGinkgoTestDescription().Duration` to know the
current duration of the test, so the helper only search in the logs
files since test starts. (In case of any panic in log, it'll not mark
all test as failed)

The panic string example is:

```
Mar 15 16:35:50 runtime cilium-agent[8220]: panic: runtime error: invalid memory address or nil pointer dereference
Mar 15 16:35:50 runtime cilium-agent[8220]: [signal SIGSEGV: segmentation violation code=0x1 addr=0x2 pc=0x17e3b2c]
Mar 15 16:35:50 runtime cilium-agent[8220]: goroutine 58384 [running]:
```

The deadlock string example is:
```
Jan 03 00:40:09 cilium-master cilium-agent[32439]: POTENTIAL DEADLOCK:
Jan 03 00:40:09 cilium-master cilium-agent[32439]: Previous place where the lock was grabbed
Jan 03 00:40:09 cilium-master cilium-agent[32439]: goroutine 8974 lock 0xc4201b875c
```

Signed-off-by: Eloy Coto <eloy.coto@gmail.com>

### CODE REVIEW

**Comment by eloycoto** [comment]
test-me-please

**Comment by eloycoto** [comment]
test-me-please

**Comment by eloycoto** [comment]
test-me-please

File: test/helpers/cilium.go [hunk 1]
```diff
@@ -527,6 +528,21 @@ func (s *SSHMeta) ReportFailed(commands ...string) {
 	fmt.Fprint(wr, "===================== EXITING REPORT GENERATION =====================\n")
 }
 
+// ValidateNoErrorsOnLogs checks in cilium logs since the given duration (By
+// default `CurrentGinkgoTestDescription().Duration`) that no `panic` messages
```
Comment (ianvernon) [inline test/helpers/cilium.go:532]: ... checks that the cilium logs ... do not contain `panic` or `deadlock` messages.

File: test/helpers/kubectl.go [hunk 2]
```diff
@@ -777,6 +778,24 @@ func (kub *Kubectl) CiliumReport(namespace string, pod string, commands ...[]str
 
 }
 
+// ValidateNoErrorsOnLogs checks in cilium logs since the given duration (By
```
Comment (ianvernon) [inline test/helpers/kubectl.go:781]: ... checks that the cilium logs ... do not contain panic or deadlock messages.

**Review by ianvernon — APPROVED** [review]
Looks great overall, just some minor doc fixes needed, so no need to re-ask for review again once you have fixed them. Thanks for doing this! One comment -in the future, please have vendor updates as separate commits.

**Review by raybejjani — APPROVED** [review]

**Comment by raybejjani** [comment]
oh, crap. I merged this without waiting for the changes ian asked for. @eloycoto could you make another PR for that and I'll merge it in. sorry

**Comment by ianvernon** [comment]
@raybejjani no worries! Nothing was blocking on my review.

**Pull request merged**